src/components/Board/index.js

```diff
@@ -59,8 +60,11 @@ function Board ({
     setBoard(reorderedBoard)
   }
 
-  async function addLane (title) {
-    const lanes = addInArrayAtPosition(board.lanes, await onNewLane({ title, cards: [] }), board.lanes.length)
+  async function addLane (lane) {
```

Comment:
What if you change the default lane adder to pass the entire lane instead of just the title? You might don't need to do the complex logic below.